### PR TITLE
Add the Stone Maze Map.

### PR DESCRIPTION
**Pull request by Jo5629**
This is my first map I have made, so there will probably be some errors. Feel free to call me out so I can polish my map-making skills a little bit more.

The ground is two blocks deep because I do not support tunneling 😅 (but you can tunnel under the maze walls)

screenshot

### CODE REVIEW

**Comment by LuismiInTheZone** [comment]
The map is actually very cool, there are some mistakes:

For example, I can go outside the glass barriers before the match starting, specially the one from the ceiling, please check again the points. If you need some help, just ask in #maps from discord. Also another thing, I don't know if is me, but reds part look lot bigger than other teams parts, check it (or wait for other testers) to see that. Nice work, looks nice

**Comment by lazylier** [comment]
This looks like a fun map. I couldn't find any holes in the glass barrier. The only thing I'd suggest changing about that is making the bottom of the map indestructible dirt so players can't dig down and stand on the glass. You might not need to include "default:pick_stone" in the initial_stuff. As for the treasures, I think maybe add an extra diamond for each team, so they have enough for two swords. 

Red's section of the map is slightly larger than the other teams'. Red and Orange have bigger base areas than the Blue and Green, and Blue's is smaller than Green's. I don't know if any of that matters.

**Comment by LuismiInTheZone** [comment]
Also another thing, I think the screenshot doesn't show the map, just the box with ores, maybe making another screenshot, I don't know, maybe from a point from one base to the start of the maze...

**Review by HobbitPower — CHANGES_REQUESTED** [review]
Interesting idea, I like it so far.
* As I can see from your second clue, the intention was to make the walls breakable. This can somewhat defeat the purpose of a Maze Map, but on the other hand, I don't think it's advisable to make everything out of unbreakable blocks. Maybe some of them could be made indestructible? Or maybe make the middle of the walls destructible and use indestructible blocks for the top & bottom layers?
* I also honestly don't know if we should enable it for nade fight, with all the high walls.
* I would advise you to use 4 chest zones instead of one, one for each team. 50 chests for 4 teams is generally very little. I would suggest more like 20-25 per team. Either way, I advocate 4 separate chest zones.
* Your screenshot should have a 3:2 aspect ratio and show the whole map. So something like the following (I took the liberty and removed the top layer of glass to create a better image). You can use this as a screenshot or create your own. The only important thing is the right aspect ratio, that no barrier is visible and also no coordinates/HUD bar etc. are visible and that you name it screenshot.png (etc.). 
screenshot


I'll take a closer look soon, good work so far.

**Comment by Jo5629** [comment]
> * As I can see from your second clue, the intention was to make the walls breakable. This can somewhat defeat the purpose of a Maze Map, but on the other hand, I don't think it's advisable to make everything out of unbreakable blocks. Maybe some of them could be made indestructible? Or maybe make the middle of the walls destructible and use indestructible blocks for the top & bottom layers?
> * I also honestly don't know if we should enable it for nade fight, with all the high walls.
> * I would advise you to use 4 chest zones instead of one, one for each team. 50 chests for 4 teams is generally very little. I would suggest more like 20-25 per team. Either way, I advocate 4 separate chest zones.
> * Your screenshot should have a 3:2 aspect ratio and show the whole map. So something like the following (I took the liberty and removed the top layer of glass to create a better image). You can use this as a screenshot or create your own. The only important thing is the right aspect ratio, that no barrier is visible and also no coordinates/HUD bar etc. are visible and that you name it screenshot.png (etc.).

My updates:

1. I made the top five and the bottom five blocks of the maze into unbreakable blocks.
2. I kept it for nade fight for now, I will take it off if requested by the general public.
3. I did 4 chest zones with 25 chests each.
4. I used your screenshot @HobbitPower.

**Review by HobbitPower — COMMENTED** [review]
@Jo5629  Not my final review:

The changes are looking good so far. Another thing that came to my mind: You could make the walls destructible only for certain block types (e.g. cobble and mossy cobble) and the rest indestructible. But I would wait for more feedback on how the walls should look like. I think it's good for now that they are partially indestructible.

I saw your credit regarding the screenshot, but if you want to note that, it rather belongs in "others" (more informations) or you leave it out completely ;) "You can break parts of the walls." is more of a hint instead of the acknowledgement ^^

And I would suggest to replace the orange team by team purple, because red and orange are a bit similar. As far as I know no 4-teams map has red and orange as teams.

But otherwise I find the map quite good and promising.

**Comment by Jo5629** [comment]
> @Jo5629 Not my final review:
> 
> The changes are looking good so far. Another thing that came to my mind: You could make the walls destructible only for certain block types (e.g. cobble and mossy cobble) and the rest indestructible. But I would wait for more feedback on how the walls should look like. I think it's good for now that they are partially indestructible.

I probably won't do that because I used //mix so the variety will be off at times.

> 
> I saw your credit regarding the screenshot, but if you want to note that, it rather belongs in "others" (more informations) or you leave it out completely ;) "You can break parts of the walls." is more of a hint instead of the acknowledgement ^^

Done.

> 
> And I would suggest to replace the orange team by team purple, because red and orange are a bit similar. As far as I know no 4-teams map has red and orange as teams.

Can I edit the map.conf file or will I have to do that ingame?

> 
> But otherwise I find the map quite good and promising.

Thank you.

**Comment by a-blob** [comment]
I didn't find any issues. I like the overall idea. However, after playing on it for a bit I have a few suggestions.

- Personally, I find the map a bit bland. The builds lack some detail. For example, it would be nice to see plants climbing the maze walls.
- Same with the ground. For now, it is just dirt. Maybe there could be some plants, grass, or/and rocks scattering among the dirt?
- Just an idea but maybe the maze design could be more interesting. For example, there could be some gaps in the walls and staircases that lead up to them (or just a few staircases leading up the walls to indicate to the player that they are breakable up there). Or there could be elevation differences in parts of the maze. Overall it would be nice to see some stairs going around the maze.
- I find the glass box of ores in the middle a bit out of place. Maybe the structure could be more interesting.

**Comment by Jo5629** [comment]
I added multiple types of dirt for the ground and updated the screenshot.

**Comment by Jo5629** [comment]
> Ultimately, the map maker decides on the design of the map. But I have to agree with GreenBlob that the glass box with the ores doesn't really fit the look of the rest of the map. You could, for example, build a small temple over it or a dome. (That was just a quick attempt on my part with WorldEdit, I'm sure you could come up with something more creative:) image

Fair enough. I will work on that.
> 
> @Jo5629 One more thing: **If** you edit your map, you should update your CTF game. The map editor has been changed and now you will receive the map.conf file and the map.mts. file as well as the new barriers.data file after you finish editing. So save your current map as a backup, update your game and edit your map in a new world. At the end, please replace the files of your PR and add the barriers.data file to it.

The only reason is that when I updated and loaded my map, it would not let me. It just gave off a zlib compress error for some reason. I will try now though and report here if anything goes wrong.

**Comment by Jo5629** [comment]
Changes I Made:

1. Changed the glass box into the pyramid (Probably needs to check for any imbalances and random nodes)
2. Got the new version of CTF to work and added the barriers.data

Should be updated to everyone's requests AFAIK

**Comment by src4026** [comment]
- Style:
image
I find that Jungle wood/Apple wood would blend in better with the surroundings as the pine wood looks a bit out of the blue in the midst of its environment currently. 
As has been mentioned by GreenBlob perviously, you could add more decoration to the walls of the map such as vines, and so on. In addition to that, I suggest that adding wall decorations reminiscent of mayan masks and carvings can also be an option. In my imagination they could fit the style of the map quite nicely.
As it seems, you have made the destructible blocks as the area bettwen five blocks from the bottom and five from the top. This could be too large though IMO, and can perhaps be reduced further as it isn't much different from the entire maze being destructible, and quoting, "This can somewhat defeat the purpose of a Maze Map". Currently it's better, but I think it would be good to narrow the area. 

> Changed the glass box into a pyramid (Probably need to check it for any ore imbalances and random nodes)

Ore counts per team:
- Green: 6 mese ores; 4 diamond ores; 6 Iron ores; 2 Coal ores
- Purple: 5 mese ores; 4 diamond ores; 6 Iron ores; 2 Coal ores
- Blue: 6 mese ores; 4 diamond ores; 6 Iron ores; 2 Coal ores
- Red: 6 mese ores; 4 diamond ores; 6 Iron ores; 2 Coal ores

As it seems, Purple only has one less mese as compared to the rest.

To sum it up, the I find the map really good and well-crafted. Waiting for further updates and replies eagerly :)

---
Note: There are some barrier removal bugs in some maps below a certain `y` value. This one seems to be among the ones affected by it. In discussion with map managers,  testers, and LoneWolfHT at the moment.

**Comment by Jo5629** [comment]
>   I find that Jungle wood/Apple wood would blend in better with the surroundings as the pine wood looks a bit out of the blue in the midst of its environment currently.

Have done that.
>   As has been mentioned by GreenBlob perviously, you could add more decoration to the walls of the map such as vines, and so on. In addition to that, I suggest that adding wall decorations reminiscent of mayan masks and carvings can also be an option. In my imagination they could fit the style of the map quite nicely.
>   I tested some areas (hopefully didn't miss any cases) and I see that you have made the destructible blocks as the area bettwen five blocks from the bottom and five from the top. This could be too large though IMO, and can perhaps be reduced further as it isn't much different from the entire maze being destructible, and quoting, "This can somewhat defeat the purpose of a Maze Map". Currently it's better, but I think it would be good to narrow the area.

Narrowed the unbreakable height to 3, so it can just be countered by a block in the middle.

> 
> > Changed the glass box into a pyramid (Probably need to check it for any ore imbalances and random nodes)
> 
> Ore counts per team:
> 
> * Green: 6 mese ores; 4 diamond ores; 6 Iron ores; 2 Coal ores
> * Purple: 5 mese ores; 4 diamond ores; 6 Iron ores; 2 Coal ores
> * Blue: 6 mese ores; 4 diamond ores; 6 Iron ores; 2 Coal ores
> * Red: 6 mese ores; 4 diamond ores; 6 Iron ores; 2 Coal ores
> 
> As it seems, Purple only has one less mese as compared to the rest.

Have changed the ores.
> 
> To sum it up, the I find the map really good and well-crafted. Waiting for further updates and replies eagerly :)
> 
> Note: There are some barrier removal bugs in some maps below a certain `y` value. This one seems to be among the ones affected by it. In discussion with map managers, testers, and LoneWolfHT at the moment.

The changes I have will be in the next few days. I have a lot of things on my plate so it will probably be not ready until about the end of this week or the middle of next week.

**Comment by src4026** [comment]
No hurries 👍

**Comment by HobbitPower** [comment]
@a-blob Could you maybe briefly test the map when you have time and tell me if the barriers disappear completely after the building time?

**Comment by a-blob** [comment]
@HobbitPower The barriers don't disappear completely. 
image

**Comment by HobbitPower** [comment]
> @HobbitPower The barriers don't disappear completely. image

@LoneWolfHT If you have time, maybe you could test this map once? It's just that at least 3 people have this problem by now. 
The same goes when testing the Water Academy map. Unlike most other maps, these two maps take place on the lowest level and so it's immediately noticeable. I think it has to do with the fact that for all maps the smaller y-coordinate for `barrier_area` in the map.conf has the value `16`, where it should probably be `y=0`. x and z coordinates seem to be correct. 

Water Academy on the server works as far as I know, so I don't know if the problem only occurs when testing locally. I also tried re-saving the maps without success.

**Comment by HobbitPower** [comment]
> > I tested some areas (hopefully didn't miss any cases) and I see that you have made the destructible blocks as the area bettwen five blocks from the bottom and five from the top. This could be too large though IMO, and can perhaps be reduced further as it isn't much different from the entire maze being destructible, and quoting, "This can somewhat defeat the purpose of a Maze Map". Currently it's better, but I think it would be good to narrow the area.
> 
> Narrowed the unbreakable height to 3, so it can just be countered by a block in the middle.

I think there is a misunderstanding here @Jo5629 If I understand @src4026 correctly, you should fill the walls with more indestructible blocks. At least that's what **I** would recommend. You already have a pattern in the walls, maybe it would be enough to replace certain block types (e.g. stone and moss pavement) with their indestructible variant. That way the players would have to find a way through the walls instead of just stacking up 3 blocks and simply breaking through. The lowest rows of blocks should still be completely indestructible, in my opinion.
As already suggested, you can of course still add decorations such as vines or implied/broken stairs leading to nowhere, cracks in the walls, masks/statues/carvings.

**Comment by src4026** [comment]
Oh right, thank you.

**Comment by olliy1or** [comment]
@Jo5629 when the map is finished, request for review

**Comment by farooqkz** [comment]
Hello. One of my wishes about having a maze map in CTF is about to become true thanks to @Jo5629!

One suggestion I have here: The maze walls must all be indestructible, then in the lua file, for each match, some of the wall blocks could be replaced with destructible blocks. Similar to what we want to do for the Tunnel map according to a suggestion. This way, it adds more fun to the game to find destructible holes. And thus more strategy.

**Comment by LoneWolfHT** [comment]
> @LoneWolfHT If you have time, maybe you could test this map once? It's just that at least 3 people have this problem by now. The same goes when testing the Water Academy map. Unlike most other maps, these two maps take place on the lowest level and so it's immediately noticeable. I think it has to do with the fact that for all maps the smaller y-coordinate for `barrier_area` in the map.conf has the value `16`, where it should probably be `y=0`. x and z coordinates seem to be correct.
> 
> Water Academy on the server works as far as I know, so I don't know if the problem only occurs when testing locally. I also tried re-saving the maps without success.

Did I ever fix this issue?

**Comment by HobbitPower** [comment]
> > @LoneWolfHT If you have time, maybe you could test this map once? It's just that at least 3 people have this problem by now. The same goes when testing the Water Academy map. Unlike most other maps, these two maps take place on the lowest level and so it's immediately noticeable. I think it has to do with the fact that for all maps the smaller y-coordinate for `barrier_area` in the map.conf has the value `16`, where it should probably be `y=0`. x and z coordinates seem to be correct.
> > Water Academy on the server works as far as I know, so I don't know if the problem only occurs when testing locally. I also tried re-saving the maps without success.
> 
> Did I ever fix this issue?

@LoneWolfHT I think so. The maps in question no longer have this problem, nor do any of the newly created PRs as far as I can see.

**Comment by Jo5629** [comment]
Have been preoccupied but I have been slowly working on the map and taking in suggestions and criticism from everyone.

**UPDATES:** 
- I have removed the mossy cobble in favor of more ladders.
- I upgraded the pickaxe from cobble to steel and chest areas like Themostrandom suggested.
- I was able to make a node and code that randomizes the walls like the idea from farooqkz.
  - Not implemented inside the map fully yet.
  - [Test map.](https://github.com/user-attachments/files/15844113/stone_maze2.zip) Unzip and do the same procedure as you would moving any map to your folders. `/ctf_next -f stone_maze2` is the command. Will not update and package with the main map until I feel fully ready.

File: stone_maze/map.conf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+teams = local _={};_[1]="enabled";_[2]="flag_pos";return {green={pos2={z=200,x=0,y=-0.5},[_[2]]={z=194,x=6,y=2.5},[_[1]]=true,pos1={z=100,x=100,y=29.5}},blue={pos2={z=0,x=0,y=-0.5},[_[2]]={z=6,x=6,y=2.5},[_[1]]=true,pos1={z=100,x=100,y=29.5}},purple={pos2={z=200,x=200,y=-0.5},[_[2]]={z=194,x=194,y=2.5},[_[1]]=true,pos1={z=100,x=100,y=29.5}},red={pos2={z=0,x=200,y=-0.5},[_[2]]={z=6,x=194,y=2.5},[_[1]]=true,pos1={z=100,x=100,y=29.5}}}
+chests = return {{pos2={z=100,x=100,y=7.5},amount=25,pos1={z=0,x=0,y=2.5}},{pos2={z=100,x=200,y=7.5},amount=25,pos1={z=0,x=100,y=2.5}},{pos2={z=200,x=200,y=7.5},amount=25,pos1={z=100,x=100,y=2.5}},{pos2={z=200,x=100,y=7.5},amount=25,pos1={z=100,x=0,y=2.5}}}
+phys_jump = 1
+phys_speed = 1.3
+size = return {z=200,x=200,y=31}
+map_version = 3
+enabled = true
+name = The Stone Maze
+phys_gravity = 1
+initial_stuff = return {"default:pick_steel","default:torch 30","default:ladder 35"}
```
Comment (Themostrandom) [inline stone_maze/map.conf:12]:
```suggestion
initial_stuff = return {"default:pick_steel","default:torch 30","default:ladder 50","default:cobble 50"}
```

Comment (HobbitPower) [reply stone_maze/map.conf:12]: Not necessary in my opinion. Ladders are perfectly adequate.

**Comment by HobbitPower** [comment]
@Jo5629 If it's up to me, you don't need to create a map version with a lua.file. I think it's enough if the walls are partly destructible and partly indestructible, as it appears in the current version of the PR. The vines are a good approach. I think the map is still in WIP, so you will add more things or at least more vines. Looks good, keep it up.
The pattern can stay the same and doesn't have to change every round. But if you want to use the init.lua feature, LandarVargan will have to take a look at it.

**Comment by ClothierEdward** [comment]
This map reminds me a lot of a maze map that was on the server a while ago but exectued much better since it is a 2D maze instead of a 3D maze.
There's only a few things I'd change with this map.
1. Make it more clear what parts of the map can be mined through. Maybe make it so that only cobblestone can be broken on this map and create some cobblestone tunnels inbetween some of the walls.
2. Get rid of the fire that's on top of the map. It looks unintentional and really out of place.
screenshot_20250202_181223

**Comment by Themostrandom** [comment]
Hi, 

I have a comment and a suggestion for your map:

Some of the blocks on the walls are breakable (cobble) and others are not (stone). Apart from that, your map seems OK, but I'd like to make a suggestion: make arrows on the ground to the centre to guide the players.

image
image
image
